File: python/perfect_numbers.py
def classify(number):
    """ A perfect number equals the sum of its positive divisors.

    :param number: int a positive integer
    :return: str the classification of the input integer
    """
    if number <= 0:
        raise ValueError('Classification is only possible for positive integers.')
    lst = [i for i in range(1, number) if number % i == 0]
    sum_ = sum(lst)
    if sum_ == number:
        return 'perfect'
    if sum_ > number:
        return 'abundant'
    return 'deficient'

File: lua/perfect-numbers.lua
local function aliquot_sum(n)
    local sum = 0
    for i = 1, n - 1 do
        if n % i == 0 then
            sum = sum + i
        end
    end
    return sum
end

local function classify(n)
    local sum = aliquot_sum(n)
    if sum < n then
        return "deficient"
    elseif sum == n then
        return "perfect"
    else
        return "abundant"
    end
end

return {
    aliquot_sum = aliquot_sum,
    classify = classify,
}
